crates/sequencer/src/sequencer.rs

```diff
@@ -922,7 +922,14 @@ where
         let mut missed_da_blocks_count = 0;
 
         loop {
-            let mut interval = tokio::time::interval(target_block_time - parent_block_exec_time);
+            // We try to produce the next block based on how long the previous block took in terms
+            // of execution time. If last block takes more than target block time (2 secs),
+            // we just reset the interval to 2 secs.
+            let mut interval = if parent_block_exec_time > target_block_time {
+                tokio::time::interval(target_block_time)
```

Comment:
if producing the parent block took more than 2 secs, shouldn't we wait 0 seconds?

Reply:
Each block gets created every 2 seconds.
If parent block takes 1.5 seconds, we produce the next block after 0.5 seconds but if the parent block takes it's whole slot we produce the next block after it's own 2 seconds.

Reply:
Though the more correct approach to have consistent 2 second blocks would be:
If parent block takes 1.5 seconds, the remainder gets added to the next block's time to become 2.5 seconds.